cpp/tests/experimental/join/quadtree_point_in_polygon_test_small.cu

```diff
@@ -0,0 +1,167 @@
+     {6.6793884701899, 2.5605928243991434},     {6.4274219368674315, 2.9754616970668213},
+     {6.444584786789386, 2.174562817047202},    {7.897735998643542, 3.380784914178574},
+     {7.079453687660189, 3.063690547962938},    {7.430677191305505, 3.380489849365283},
+     {7.5085184104988, 3.623862886287816},      {7.886010001346151, 3.538128217886674},
+     {7.250745898479374, 3.4154469467473447},   {7.769497359206111, 3.253257011908445},
+     {1.8703303641352362, 4.209727933188015},   {1.7015273093278767, 7.478882372510933},
+     {2.7456295127617385, 7.474216636277054},   {2.2065031771469, 6.896038613284851},
+     {3.86008672302403, 7.513564222799629},     {1.9143371250907073, 6.885401350515916},
+     {3.7176098065039747, 6.194330707468438},   {0.059011873032214, 5.823535317960799},
+     {3.1162712022943757, 6.789029097334483},   {2.4264509160270813, 5.188939408363776},
+     {3.154282922203257, 5.788316610960881}});
+
+  // build a quadtree on the points
+  auto [point_indices, quadtree] = cuspatial::quadtree_on_points(
+    points.begin(), points.end(), v_min, v_max, scale, max_depth, max_size, this->stream());
+
+  auto poly_offsets = make_device_vector<uint32_t>({0, 1, 2, 3, 4});
+  auto ring_offsets = make_device_vector<uint32_t>({0, 4, 10, 14, 19});
+
+  auto poly_points = make_device_vector<vec_2d<T>>({// ring 1
+                                                    {2.488450, 5.856625},
+                                                    {1.333584, 5.008840},
+                                                    {3.460720, 4.586599},
+                                                    {2.488450, 5.856625},
+                                                    // ring 2
+                                                    {5.039823, 4.229242},
+                                                    {5.561707, 1.825073},
+                                                    {7.103516, 1.503906},
+                                                    {7.190674, 4.025879},
+                                                    {5.998939, 5.653384},
+                                                    {5.039823, 4.229242},
+                                                    // ring 3
+                                                    {5.998939, 1.235638},
+                                                    {5.573720, 0.197808},
+                                                    {6.703534, 0.086693},
+                                                    {5.998939, 1.235638},
+                                                    // ring 4
+                                                    {2.088115, 4.541529},
+                                                    {1.034892, 3.530299},
+                                                    {2.415080, 2.896937},
+                                                    {3.208660, 3.745936},
+                                                    {2.088115, 4.541529}});
+
```

Comment:
Can `make_multipolygon_array` help you simplify this? (You might have to materialize the geometry offset array).

https://github.com/rapidsai/cuspatial/blob/515e0c4e70374dade0ed3fe6c88df3b06b697bed/cpp/include/cuspatial_test/vector_factories.cuh#L161